lib/hls/hls_parser.js

```diff
@@ -369,6 +369,19 @@ shaka.hls.HlsParser = class {
 
     const mediaSequenceToStartTime =
         this.getMediaSequenceToStartTimeFor_(streamInfo);
+    const {keyIds, drmInfos} = this.parseDrmInfo_(playlist, stream.mimeType);
+
+    const keysAreEqual =
+      (a, b) => a.size === b.size && [...a].every((value) => b.has(value));
+
+    if (!keysAreEqual(stream.keyIds, keyIds)) {
+      stream.keyIds = keyIds;
+      stream.drmInfos = drmInfos;
+      if (this.manifest_) {
+        this.playerInterface_.filter(this.manifest_);
```

Comment:
We just added a method to the player interface that may be more appropriate for this.  Please merge main into your branch.  Then instead of filtering the entire manifest on every stream, try this:

```js
  this.playerInterface_.newDrmInfo(stream);
```